pkg/storage/stores/index/seriesvolume/volume.go

```diff
@@ -10,29 +11,36 @@ import (
 const (
 	MatchAny     = "{}"
 	DefaultLimit = 100
+
+	ErrVolumeMaxSeriesHit = "the query hit the max number of series limit (limit: %d series)"
 )
 
 // TODO(masslessparticle): Lock striping to reduce contention on this map
 type Accumulator struct {
-	lock    sync.RWMutex
-	volumes map[string]uint64
-	limit   int32
+	lock            sync.RWMutex
+	volumes         map[string]uint64
+	limit           int32
+	volumeMaxSeries int
 }
 
-func NewAccumulator(limit int32) *Accumulator {
+func NewAccumulator(limit int32, maxSize int) *Accumulator {
 	return &Accumulator{
-		volumes: make(map[string]uint64),
-		limit:   limit,
+		volumes:         make(map[string]uint64),
```

Comment:
would there be a benefit to allocating `volumes` to a capacity of `volumeMaxSeries`?